### ISSUE TITLE
Could you, please, add any screenshot or video?

### ISSUE THREAD

**Issue by egslava** [aaronrenner/zone-meal-tracker#11]
Hello! I'm trying to get now trough several hundreds of repositories to find something interesting :) It's really hard (almost impossible with my resources) to try everything. May I ask you to make some screenshots or screen recordings?

**Comment by egslava** [comment]
May be some input/output examples? Or whatever that might describe the final result :) I can see it has 35 stars, so, probably, it's something that the world should know about :)

**Comment by aaronrenner** [comment]
Hi @egslava! This repo is related to this blog post I wrote a couple of years ago: https://aaronrenner.io/2019/09/18/application-layering-a-pattern-for-extensible-elixir-application-design.html. Its goal is to show the idea of building layers of APIs in your application. There's really very little functionality here, it's more of a code architecture example.

Another note on this blog post... I made quite a bit of reference to making layers swappable, but this is something I do sparingly now. Now instead of testing layers in isolation, I'll write integration tests over unit tests to ensure everything functions as expected. I've also started using [bypass](https://hex.pm/packages/bypass) to write simulators for external services so I can ensure my application responds as I want when a connection is down or simulate a weird state in the backend service.

I hope this helps. Let me know if you have any questions.
